File: 00-playground/21-encode.py
# python3 21-encode.py "abcdefg"

import sys

# Get the input text from the command line arguments
input_text = sys.argv[1]

# Shift every character one character to the left
output_text = ""
for char in input_text:
    if char == "a":  # Handle special case for 'a'
        output_text += "z"
    elif char == "A":  # Handle special case for 'A'
        output_text += "Z"
    else:
        output_text += chr(ord(char) - 1)

# Output the result in the console
print(output_text)

File: 00-playground/21-encode.js
// node 21-encode.js "abcdefg"

// Get the input text from the command line arguments
const inputText = process.argv[2];

// Shift every character one character to the left
let outputText = "";
for (let i = 0; i < inputText.length; i++) {
  const char = inputText.charAt(i);
  if (char === "a") {
    // Handle special case for 'a'
    outputText += "z";
  } else if (char === "A") {
    // Handle special case for 'A'
    outputText += "Z";
  } else {
    outputText += String.fromCharCode(char.charCodeAt(0) - 1);
  }
}

// Output the result in the console
console.log(outputText);
